src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritConnection.java

```diff
@@ -336,18 +337,39 @@ private void nullifyWatchdog() {
     private String getLine(CharBuffer cb) {
         String line = null;
         int pos = cb.position();
+        int limit = cb.limit();
         cb.flip();
         for (int i = 0; i < cb.length(); i++) {
             if (cb.charAt(i) == '\n') {
-                line = getSubSequence(cb, 0, i).toString().trim();
+                line = getSubSequence(cb, 0, i).toString();
                 cb.position(i + 1);
                 break;
             }
         }
         if (line != null) {
             cb.compact();
+            if (eventBuffer != null) {
+                eventBuffer.append(line);
+                String eventString = eventBuffer.toString();
+                eventBuffer = null;
+                line = eventString;
+            }
+            line.trim();
         } else {
-            cb.clear().position(pos);
+            if (cb.length() > 0) {
+                if (cb.length() == cb.capacity()) {
+                    if (eventBuffer == null) {
+                        logger.debug("Encountered big event.");
+                        eventBuffer = new StringBuilder();
+                    }
+                    eventBuffer.append(getSubSequence(cb, 0, pos));
```

Comment:
It is not effective to use String Builder in such way. It would be better to push the entire chunk once you hit the limit